### PR TITLE
Cypher: SKIP/LIMIT backend support

### PR DESCRIPTION
**Pull request by l-hoang**
Skip and limit are now taken into account when query engine prints
results. Added a warning about ORDER BY not being considered as well.

### CODE REVIEW

File: libquery/include/galois/graphs/AttributedGraph.h [hunk 1]
```diff
@@ -204,18 +204,14 @@ class AttributedGraph {
...
       size_t num_matches);
 
   //! Constructs a query graph and conducts matching on the data graph.
-  //!
-  //! TODO doxygen parameters
-  size_t matchQuery(std::vector<galois::CompilerQueryNode>& queryNodes,
-                    std::vector<galois::CompilerQueryEdge>& queryEdges,
-                    const galois::gstl::Vector<QueryProperty>& return_values,
-                    std::vector<std::string>& filters, bool print_results);
+  size_t matchQuery(galois::CypherCompiler& compiler, bool print_results);
```
Comment (roshandathathri) [inline libquery/include/galois/graphs/AttributedGraph.h:214]: Fine for now but this is not a good design. The compiler should return an IR and the query engine should just read the IR. We will be discussing the IR and the engine more later, so fine for now.

File: libquery/src/AttributedGraph.cpp [hunk 2]
```diff
@@ -486,9 +482,39 @@ void AttributedGraph::printResults(
   }
   galois::gPrint("\n", divider);
 
+  // TODO int seems a bit low.....
+  // I guess the idea is that skip and limit wouldn't be used with large numbers
+  // anyways
+  const std::optional<int>& num_to_skip  = return_metadata.return_skip;
+  const std::optional<int>& result_limit = return_metadata.return_limit;
```
Comment (roshandathathri) [inline libquery/src/AttributedGraph.cpp:489]: Negative numbers for skip and limit do not make sense, right? If so, use uint32_t.

Comment (l-hoang) [reply libquery/src/AttributedGraph.cpp:489]: Parser returns them as an int

File: libquery/include/galois/graphs/AttributedGraph.h (hunk 1 above)
Comment (l-hoang) [reply libquery/include/galois/graphs/AttributedGraph.h:214]: The query engine shouldn't be constructing the query graph either for what it's worth; that should be done before it gets to this point in my opinion.

Comment (roshandathathri) [reply libquery/include/galois/graphs/AttributedGraph.h:214]: The pattern matcher component should take a pattern. i.e., query graph, as input. I think that's what you meant. The query engine that calls this component should construct the query graph.

Comment (l-hoang) [reply libquery/include/galois/graphs/AttributedGraph.h:214]: yes, the pattern matcher shuldn't be responsible for query graph construction

**Pull request merged**